Riot/Modules/Application/LegacyAppDelegate.m

```diff
@@ -1742,33 +1742,36 @@ - (void)parseUniversalLinkFragment:(NSString*)fragment outPathParams:(NSArray<NS
     *outQueryParams = queryParams;
 }
 
-
-- (BOOL)handleServerProvionningLink:(NSURL*)link
+/**
+ Parse and handle a server provisioning link. Returns `YES` if a provisioning link was detected and handled.
+ @param link A link such as https://mobile.element.io/?hs_url=matrix.example.com&is_url=identity.example.com
+ */
+- (BOOL)handleServerProvisioningLink:(NSURL*)link
 {
-    MXLogDebug(@"[AppDelegate] handleServerProvionningLink: %@", link);
+    MXLogDebug(@"[AppDelegate] handleServerProvisioningLink: %@", link);
 
     NSString *homeserver, *identityServer;
-    [self parseServerProvionningLink:link homeserver:&homeserver identityServer:&identityServer];
+    [self parseServerProvisioningLink:link homeserver:&homeserver identityServer:&identityServer];
 
     if (homeserver)
     {
         if ([MXKAccountManager sharedManager].activeAccounts.count)
         {
-            [self displayServerProvionningLinkBuyAlreadyLoggedInAlertWithCompletion:^(BOOL logout) {
+            [self displayServerProvisioningLinkBuyAlreadyLoggedInAlertWithCompletion:^(BOOL logout) {
 
-                MXLogDebug(@"[AppDelegate] handleServerProvionningLink: logoutWithConfirmation: logout: %@", @(logout));
+                MXLogDebug(@"[AppDelegate] handleServerProvisioningLink: logoutWithConfirmation: logout: %@", @(logout));
                 if (logout)
                 {
                     [self logoutWithConfirmation:NO completion:^(BOOL isLoggedOut) {
-                        [self handleServerProvionningLink:link];
+                        [self handleServerProvisioningLink:link];
                     }];
                 }
             }];
         }
         else
         {
-            [_masterTabBarController showAuthenticationScreen];
-            [_masterTabBarController.authViewController showCustomHomeserver:homeserver andIdentityServer:identityServer];
+            [_masterTabBarController showOnboardingFlow];
+            [_masterTabBarController.onboardingCoordinatorBridgePresenter showCustomHomeserver:homeserver andIdentityServer:identityServer];
```

Comment:
Currently these servers are stored if the flow hasn't hit authentication yet rather than immediately shown.